File: 3345-smallest-divisible-digit-product-i/3345-smallest-divisible-digit-product-i.java
class Solution {

    public boolean productOfDigits(int num,int t){
        int product = 1;

        while(num!=0){
            int rem = num%10;
            product = product*rem;
            num /= 10;
        }

        if(product%t ==0) return true;
        return false;
    }
    public int smallestNumber(int n, int t) {
        
        
        for(int i=0;i<10;i++){
            if(productOfDigits(n+i,t)==true){
                return n+i;
            }
        }
        return -1;
    }
}

File: 3345-smallest-divisible-digit-product-i/3345-smallest-divisible-digit-product-i.cpp
class Solution {
public:

    int productDigits(int x){
        int ans = 1;
        while(x!=0){
            int rem = x%10;
            ans *= rem;
            x/=10;
        }
        return ans;
    }
    int smallestNumber(int n, int t) {
        
        for(int i=n;i<=n+10;i++){
            if(productDigits(i)%t==0) return i;
        }
        return 0;
    }
};